src/main/java/seedu/address/model/Model.java

```diff
@@ -129,4 +130,11 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredTaskList(Predicate<Task> predicate);
+
+    /**
+     * Updates the sort order of the filtered task list using the given {@code comparator}.
+     *
+     * @throws NullPointerException if {@code comparator} is null.
+     */
+    void updateFilteredTaskList(Comparator<Task> comparator);
```

Comment:
Maybe naming it updateSortedTaskComparator or updateSortingCriteria would be better ?

Reply:
updated